src/components/controls/AudioMeter.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+
+export interface IAudioMeterRenderer {
+    (audioMonitor: any): JSX.Element | null | false;
+}
+
+export interface IAudioMeterProps {
+    audioMonitor: any;
+    render?: IAudioMeterRenderer
+};
+
+export class AudioMeter extends React.PureComponent<IAudioMeterProps, {}> {
+    private audioMeter: HTMLDivElement;
+
+    constructor(props: IAudioMeterProps) {
+        super(props);
+
+        this.refMeter = this.refMeter.bind(this);
+    }
+
+    public componentDidMount() {
+        const { audioMonitor } = this.props;
+
+        audioMonitor.on('volume_change', (volume: number) => {
+            if (this.audioMeter) {
+                // NOTE(yunsi): volume ranges from -100 to 0.
+                // TODO(yunsi): Find a better algorithm to convert db to gain.
+                const audioGain = Math.pow(10, (volume / 20));
+                this.audioMeter.style.width = 1000 * audioGain + 'px';
```

Comment:
I think we can just use `volume + 100 + '%'` and use % instead `px` for a more flexible solution

Reply:
+1

Reply:
volume is in Decibel (dB), we need to convert it to float value.
http://www.playdotsound.com/portfolio-item/decibel-db-to-float-value-calculator-making-sense-of-linear-values-in-audio-tools/